### PR TITLE
Add isUnique solution & test

### PR DESCRIPTION
**Pull request by crunchbang**

### CODE REVIEW

File: problems/datastructures/arrays-and-strings/1.1-is_uniquev2_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsUnique2(t *testing.T) {
+	testFns := []func(string) bool{isUniqueBrute, isUniqueSorted, isUniqueSimple, isUniqueOptimized}
+	for _, fn := range testFns {
+		assert.Equal(t, true, fn("heo"), "CASE heo")
```
Comment (sarathsp06) [inline problems/datastructures/arrays-and-strings/1.1-is_uniquev2_test.go:12]: Could you please   use subtest in the loop

```
        t.Run(t.Name. func(t *testing.T) {
....
}
```

https://blog.golang.org/subtests

Comment (crunchbang) [reply problems/datastructures/arrays-and-strings/1.1-is_uniquev2_test.go]: Updated.

File: problems/datastructures/arrays-and-strings/1.2-check_permutation.go [hunk 2]
```diff
@@ -20,3 +22,57 @@ func isPermutation(s1, s2 string) bool {
...
+	return sortString(s1) == sortString(s2)
+}
+
+// O(n) : Use maps to keep a character count
+// Technically
+// O(n) + O(n) + constant number of map keys = O(n)
+// Alternatively, you can use an array instead of a map since there
+// are only a fixed number of characters in the alphabet.
+func isPermutationMap(s1, s2 string) bool {
+	if !isSameLength(s1, s2) {
```
Comment (sarathsp06) [inline problems/datastructures/arrays-and-strings/1.2-check_permutation.go:41]: personally i would not use a function for it as the level of abstraction in this function  allows the code from inside that to be here

Comment (crunchbang) [reply problems/datastructures/arrays-and-strings/1.2-check_permutation.go:41]: True. I just took DRY a bit too far. :P

**Pull request merged**